mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionModelUtils.java

```diff
@@ -230,6 +231,44 @@ private void resolve(ComponentModel model, Object owner) {
     return result.get();
   }
 
+  /**
+   * @param extensionModel the model which owns the {@code component}
+   * @param component a component
+   * @return The config (if any) that the {@code component} needs in order to function
+   */
+  public static Optional<ConfigurationModel> getConfigurationFromComponent(ExtensionModel extensionModel, NamedObject component) {
+    if (!(component instanceof ComponentModel)) {
+      return empty();
+    }
+
+    Reference<Object> result = new Reference<>();
+
+    new ExtensionWalker() {
+
+      @Override
+      public void onOperation(HasOperationModels owner, OperationModel model) {
+        resolve(model, owner);
+      }
+
+      @Override
+      public void onSource(HasSourceModels owner, SourceModel model) {
+        resolve(model, owner);
+      }
+
+      private void resolve(ComponentModel model, Object owner) {
+        if (model == component) {
+          if (!(owner instanceof ExtensionModel)) {
+            result.set(owner);
+          } else if (model.requiresConnection()) {
+            extensionModel.getConfigurationModel(DEFAULT_CONFIG_NAME).ifPresent(result::set);
```

Comment:
it is imposible for the model to require a connection if it doesn't have a config